liblocation/src/main/java/com/mapbox/android/core/location/GoogleLocationEngine.java

```diff
@@ -2,26 +2,34 @@
 
 import android.content.Context;
 import android.location.Location;
+import com.google.android.gms.location.LocationListener;
 import android.os.Bundle;
+import android.os.Looper;
 import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
 
 import com.google.android.gms.common.ConnectionResult;
 import com.google.android.gms.common.api.GoogleApiClient;
-import com.google.android.gms.location.LocationListener;
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.LocationCallback;
 import com.google.android.gms.location.LocationRequest;
+import com.google.android.gms.location.LocationResult;
 import com.google.android.gms.location.LocationServices;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
 
 import java.lang.ref.WeakReference;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 /**
  * Sample LocationEngine using Google Play Services
  */
-class GoogleLocationEngine extends LocationEngine implements
-  GoogleApiClient.ConnectionCallbacks, GoogleApiClient.OnConnectionFailedListener, LocationListener {
+class GoogleLocationEngine extends LocationEngine implements GoogleApiClient.ConnectionCallbacks,
+  GoogleApiClient.OnConnectionFailedListener, LocationListener {
 
+  private FusedLocationProviderClient fusedLocationProviderClient;
   private WeakReference<Context> context;
   private GoogleApiClient googleApiClient;
```

Comment:
https://android-developers.googleblog.com/2017/06/reduce-friction-with-new-location-apis.html

shows that GoogleApiClient shouldn't be used anymore. Please remove